jscomp/common/js_config.ml

```diff
@@ -23,79 +23,45 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA. *)
 
-
-
-
-
-
-
 (** Browser is not set via command line only for internal use *)
 
-
 let no_version_header = ref false
-
 let cross_module_inline = ref false
-
-
-
 let diagnose = ref false
-let get_diagnose () = 
+
+let get_diagnose () =
   !diagnose
   
-# 44 "common/js_config.pp.ml"
+# 34 "common/js_config.pp.ml"
   || Sys.getenv_opt "RES_DEBUG_FILE" <> None
 
-# 47 "common/js_config.pp.ml"
+# 37 "common/js_config.pp.ml"
 (* let (//) = Filename.concat *)
 
 (* let get_packages_info () = !packages_info *)
 
 let no_builtin_ppx = ref false
-
-
-
-
 let tool_name = "ReScript"
-
 let check_div_by_zero = ref true
 let get_check_div_by_zero () = !check_div_by_zero
-
-
-
 let syntax_only = ref false
 let binary_ast = ref false
-
-
-
 let debug = ref false
-
-let cmi_only = ref false  
+let cmi_only = ref false
 let cmj_only = ref false
-
 let force_cmi = ref false
 let force_cmj = ref false
-
 let jsx_version = ref (-1)
```

Comment:
Why is this the default. I thought V3 would be the default. For backwards compatibility.
Or is `-1` going to behave like V3?

Reply:
The default -1 is for the case that `jsx.version` config is not existing.

Side comment: If `jsx.version` config is not existing, then ppx should not running at the beginning. I think I should fix this https://github.com/mattdamon108/rescript-compiler/blob/4a9f7c609c72bc9fb37efb6a95016792df63aa58/jscomp/frontend/ppx_entry.ml#L30, checking `Js_config.jsx_version` here and decide to call ppx or not at the very begining.

Reply:
> Side comment: If `jsx.version` config is not existing, then ppx should not running at the beginning. I think I should fix this https://github.com/mattdamon108/rescript-compiler/blob/4a9f7c609c72bc9fb37efb6a95016792df63aa58/jscomp/frontend/ppx_entry.ml#L30, checking `Js_config.jsx_version` here and decide to call ppx or not at the very begining.

Isn't this a breaking change?
Most existing projects don't have any config for react ppx.

Reply:
> > Side comment: If `jsx.version` config is not existing, then ppx should not running at the beginning. I think I should fix this https://github.com/mattdamon108/rescript-compiler/blob/4a9f7c609c72bc9fb37efb6a95016792df63aa58/jscomp/frontend/ppx_entry.ml#L30, checking `Js_config.jsx_version` here and decide to call ppx or not at the very begining.
> 
> Isn't this a breaking change? Most existing projects don't have any config for react ppx.

Actually I was wrong. This was my assumption but it is not true.
Go ahead with your plan.

Reply:
The only suggestion is to make the version a variant (including DoNotRun) instead of a number. Convert into it at config parsing time.

Reply:
Not sure I understand. Currently numbers are used. This question is whether another representation has been considered. Sure it's possible to convert numbers back and forth to variants of the form e.g. `V3 | V4 | NotSelected` from `3`, `4`, or any other number.

Reply:
That makes sense. I meant that it seems little bit redundant to parse and convert values back and forth, because `Js_config.*` values are being used only for triggering ppx or not in `ppx_entry.ml`. I change my mind. It is usually better to make the value into variants as a data modeling, especially for this kind of sum type values. I'll fix it.

Reply:
Forget my terminology. I meant just convert once on parsing. And possibly give error if the value is not in the acceptable range.

Reply:
It really is not a deep comment. Just, if we can avow having `-1` in the code, why not avoid it.

Reply:
I did data modeling again with `jsx_version option ref`. I think best way to avoid `-1` and better data modelling is `type jsx_version = Jsx_v3 | Jsx_v4` without `NotSelected`. Does it make sense to you?

Reply:
sure that's the same as NotSelected.